File: CT/Dev/1.java
class Solution {
    public int solution(int[] d, int m) {
        int answer = 0;
        int boxes = 1;
        
        for (int num:d) {
            answer += 1;
            if (num >= boxes) {
                m -= boxes;
                boxes *= 2;
            } else {
                boxes = 1;
            }
            if (m <= 0) {
                break;
            }
        }
        if (m > 0) {
            answer = -1;
        }
        return answer;
    }
}

File: CT/WTC2021/1.py
from collections import defaultdict

def solution(arr):
    answer = []
    num_dict = {1: 0, 2: 0, 3: 0}

    for num in arr:
        num_dict[num] += 1

    max_cnt = max(num_dict.values())

    for cnt in num_dict.values():
        answer.append(max_cnt - cnt)

    return answer